raiwidgets/tests/test_cohort.py

```diff
@@ -141,7 +150,343 @@ def test_cohort_filter_serialization_include_exclude_methods(self, method):
+                                                          method)):
+            cohort_filter_regression = \
+                CohortFilter(method=method,
+                             arg=[2.5],
+                             column=target_filter_type)
+            cohort_filter_regression._validate_with_test_data(
+                test_data=test_data_regression,
+                target_column="target",
+                categorical_features=[],
+                is_classification=False)
+
+    @pytest.mark.parametrize('target_filter_type',
+                             [CohortFilter.PREDICTED_Y,
+                              CohortFilter.TRUE_Y])
+    def test_validate_with_test_data_classification_target_filter_validations(
+            self, target_filter_type):
+        test_data_classification = pd.DataFrame(
+            data=[[23, 'A'], [25, 'B'], [25, 'B']],
+            columns=["age", "target"])
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="{0} can only be configured with "
+                      "filter {1} for classification".format(
+                          target_filter_type,
+                          CohortFilterMethods.METHOD_INCLUDES)):
+            cohort_filter_classification = \
+                CohortFilter(method=CohortFilterMethods.METHOD_EXCLUDES,
+                             arg=['A'],
+                             column=target_filter_type)
+            cohort_filter_classification._validate_with_test_data(
+                test_data=test_data_classification,
+                target_column="target",
+                categorical_features=[],
+                is_classification=True)
+
+        with pytest.raises(
+            UserConfigValidationException,
+            match="Found a class in arg which is not present in "
+                  "test data"):
+            cohort_filter_classification = \
+                CohortFilter(method=CohortFilterMethods.METHOD_INCLUDES,
+                             arg=['Z'],
+                             column=target_filter_type)
+            cohort_filter_classification._validate_with_test_data(
+                test_data=test_data_classification,
+                target_column="target",
+                categorical_features=[],
+                is_classification=True)
+
+    def test_validate_with_test_data_with_dataset_validations(
+            self):
+        test_data = pd.DataFrame(
+            data=[[23, 'new', 'A'], [25, 'new, ''B'], [25, 'old', 'B']],
+            columns=["age", 'type', "target"])
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="{0} is a categorical feature and should be only "
+                      "configured with {1} cohort filter.".format(
+                          "type",
+                          CohortFilterMethods.METHOD_INCLUDES)):
+            cohort_filter = \
```

Comment:
I think we should try to avoid using \
I wonder if the code could be less nested by refactoring some of the pytest.raises into helper functions, not sure.
Or maybe the cohort_filter could be defined prior to the pytest.raises.

Reply:
reduced the use of \ throughout the this file.